### ISSUE TITLE
Set setNotFailingStatuses for jenkins in CucumberReportPublisher

### ISSUE THREAD

**Issue by AyushRawat1** [jenkinsci/cucumber-reports-plugin#376]
Hi @damianszczepanik,
In an issue (https://github.com/jenkinsci/cucumber-reports-plugin/issues/364), The issue was fixed but it can only be fixed programmatically is there any way we can set the non failing statuses in jenkins if not can this be fixed from your end?

Like we do for other features 
            mergeFeaturesWithRetest: true,
            sortingMethod: 'ALPHABETICAL',

can it be done for the notfailingstatuses like below??
            notFailingStatuses: 'SKIPPED'

**Comment by damianszczepanik** [comment]
I don't think this was promoted from https://github.com/damianszczepanik/cucumber-reporting project
Would you like to add support for this?

**Comment by AyushRawat1** [comment]
Hi @damianszczepanik,
yesy this issue is promoted in cucumber-reports-plugin and my point being (CucumberReportPublisher.java https://github.com/jenkinsci/cucumber-reports-plugin/blob/master/src/main/java/net/masterthought/jenkins/CucumberReportPublisher.java) is the class responsible for exposing features to jenkins right??

can we add the set not failing statuses to this class so that we can set this feature from jenkins like for other features we do  mergeFeaturesWithRetest: true,
sortingMethod: 'ALPHABETICAL' in the jenkins 

can the same be added to the CucumberReportPublisher so that i can set the not failing statuses from jenkins??

**Comment by damianszczepanik** [comment]
That's correct

**Comment by AyushRawat1** [comment]
@damianszczepanik thanks

**Comment by AyushRawat1** [comment]
@damianszczepanik any updates on this fix??

**Comment by damianszczepanik** [comment]
Sorry, I though you will be the one who promote support for this :)

**Comment by AyushRawat1** [comment]
@damianszczepanik ok let me look into it and i will support the fix

**Comment by AyushRawat1** [comment]
@damianszczepanik i currently don't have access to create PR for the issue but i have updated the feature in CucumberReportPublisher.java and i am attaching the class you can check if that's ok 

[CucumberReportPublisher.zip](https://github.com/jenkinsci/cucumber-reports-plugin/files/5291837/CucumberReportPublisher.zip)

**Comment by AyushRawat1** [comment]
@damianszczepanik and this is how i am thinking of using it through jekins, you can tell me if that's ok

  def s = ["SKIPPED"] as Set
  
    always {
        cucumber fileIncludePattern: "${env.BUILD_NUMBER}/build/cucumber/bbt_*.json",
            reportTitle: "${params.title}${env.TEST_ENVIRONMENT}/${params.cucumber_tag}",
            setNotFailingStatues: s
}

**Comment by AyushRawat1** [comment]
@damianszczepanik any review on the fix i did for the CucumberReportPublisher.java?

**Comment by damianszczepanik** [comment]
I won't do it without PR which can be created after forking the repo

**Comment by AyushRawat1** [comment]
@damianszczepanik thanks will do that

**Comment by AyushRawat1** [comment]
Hi @damianszczepanik

this is the PR for this issue that i have raised you can review it now

https://github.com/jenkinsci/cucumber-reports-plugin/pull/379/commits/caaa45ba0d71668b77b228508056a70b9474cd4d

**Comment by AyushRawat1** [comment]
@damianszczepanik can you please review the PR and tell me if anything is to be modified or changed??
Thanks

**Comment by madalinaptj** [comment]
Any updates on this? Looking for the same functionality.
